test/export/dashboard.test.ts

```diff
@@ -0,0 +1,159 @@
+                    { pollStep: 1 },
+                )
+                .then(() => {
+                    const [, options] = fetchMock.lastCall(endpoint);
+                    expect(options.method).toBe("POST");
+                    expect(JSON.parse(options.body as string)).toEqual({
+                        dashboardExport: {
+                            dashboardUri: "/gdc/md/testProjectId/obj/dashboard123",
+                            filters: [
+                                {
+                                    dateFilter: {
+                                        type: "relative",
+                                        from: "-11",
+                                        to: "11",
+                                        granularity: "GDC.time.month",
+                                    },
+                                },
+                                {
+                                    attributeFilter: {
+                                        displayForm: "/gdc/md/testProjectId/obj/700",
+                                        negativeSelection: false,
+                                        attributeElements: [
+                                            "/gdc/md/testProjectId/obj/750",
+                                            "/gdc/md/testProjectId/obj/751",
+                                        ],
+                                    },
+                                },
+                                {
+                                    dateFilter: {
+                                        type: "absolute",
+                                        from: "2019-08-06",
+                                        to: "2019-08-08",
+                                        granularity: "GDC.time.month",
+                                    },
+                                },
+                            ],
+                        },
+                    });
+                });
+        });
+
+        it("should return pdf uri", () => {
+            return dashboardModuleMock()
+                .exportDashboardToPdf(projectId, dashboardUri, [], { pollStep: 1 })
+                .then((result: IExportResponse) => {
+                    expect(result).toEqual({
+                        uri: "/gdc/exporter/result/testProjectId/pdfFileUri",
+                    });
+                });
+        });
+    });
+
+    describe("failed pdf export", () => {
+        it("should return error when send export request fail", () => {
+            fetchMock.post(endpoint, {
+                status: BAD_REQUEST_STATUS,
+            });
+
+            return dashboardModuleMock()
+                .exportDashboardToPdf(projectId, dashboardUri, [])
+                .then(null, (error: ApiResponseError) => {
+                    // error thrown in xhr.ts
+                    expect(error.response.status).toEqual(400);
+                    expect(error.message).toEqual("Bad Request");
```

Comment:
Please add constant for `Bad Request`

Reply:
Could you please explain why?
I think this is raw mock data, so we can define it directly.
And in case we define new, should it be only in this local test file? I don't use it anywhere else in my impl though.

Reply:
We already have this in errors.ts
`export const BAD_REQUEST_MESSAGE = "Bad Request";`
Reuse this could help the code easier to maintain later.

The current texts are ok to me